test/libsolidity/ASTJSONTest.cpp

```diff
@@ -69,20 +94,30 @@ void replaceTagWithVersion(string& _input)
 
 }
 
-
-ASTJSONTest::ASTJSONTest(string const& _filename)
+void ASTJSONTest::generateTestVariants(string const& _filename)
 {
-	if (!boost::algorithm::ends_with(_filename, ".sol"))
-		BOOST_THROW_EXCEPTION(runtime_error("Invalid test contract file name: \"" + _filename + "\"."));
-
 	string_view baseName = _filename;
 	baseName.remove_suffix(4);
 
-	m_variants = {
-		TestVariant(baseName, CompilerStack::State::Parsed),
-		TestVariant(baseName, CompilerStack::State::AnalysisPerformed),
+	const std::vector<CompilerStack::State> variantCompileStates = {
+		CompilerStack::State::Parsed,
+		CompilerStack::State::AnalysisPerformed
 	};
 
+	for (const auto state: variantCompileStates)
```

Comment:
```suggestion
	for (CompilerStack::State state: {
		CompilerStack::State::Parsed,
		CompilerStack::State::AnalysisPerformed
	})
```
or otherwise: the coding style is ``<type> const`` not ``const <type>`` ;-).